### PR TITLE
Purge PETSc from IO code

### PR DESCRIPTION
**Pull request by garth-wells**

### CODE REVIEW

File: cpp/dolfinx/io/xdmf_function.h [hunk 1]
```diff
@@ -30,7 +30,11 @@ namespace xdmf_function
 {
 
 /// TODO
-void add_function(MPI_Comm comm, const fem::Function<PetscScalar>& u,
+void add_function(MPI_Comm comm, const fem::Function<double>& u, const double t,
+                  pugi::xml_node& xml_node, const hid_t h5_id);
+
+/// TODO
```
Comment (chrisrichardson) [inline cpp/dolfinx/io/xdmf_function.h:36]: What is TODO?

**Comment by chrisrichardson** [comment]
I've been keen on doing this for a while... can we now mix I/O of real and complex in the same build then?

File: cpp/dolfinx/io/xdmf_function.h (hunk 1 above)
Comment (garth-wells) [reply cpp/dolfinx/io/xdmf_function.h:36]: No idea it - was already there.

**Comment by garth-wells** [comment]
There is one tiny issue with `FiniteElement` and `ufc_scalar_type_t` still.

**Pull request merged**